Unified Sphinx into one.

agcapps opened the pull request:
Use `make docs` to build doxygen and sphinx.

agcapps commented:
Use `make doxygen_docs` or `make sphinx_docs` to build just those parts; use `make docs` to build everything.

agcapps commented:
The theme is broken; needs fixing.

agcapps commented on src/axom/slic/docs/sphinx/index.rst:
```diff
@@ -0,0 +1,9 @@
+.. ## Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
```
@gzagaris , I know you have the slic docs PR in progress.  This is a placeholder.

gzagaris replied:
Thanks. I'll rebase my branches based on this.

agcapps commented on src/conf.py:
```diff
@@ -73,7 +93,7 @@
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
-exclude_patterns = ['_build']
+exclude_patterns = [ '_build', 'cmake/blt/docs' ]
```
This excludes the BLT docs that exist under this directory but aren't linked in to the toctree.

agcapps commented on src/conf.py:
```diff
@@ -43,7 +63,7 @@
 
 # The master toctree document.
-master_doc = 'index'
+master_doc = 'docs/index'
```
web main index.rst and associated files now sit directly in the docs directory.

agcapps commented on src/conf.py:
```diff
@@ -285,3 +311,11 @@
+
+# add breathe support
+
+breathe_projects = { "axomdoc": "@CMAKE_CURRENT_BINARY_DIR@/../doxygen/xml/" }
```
I think this should be built up the way that `html_static_path` is.  @white238 , can you confirm this?

white238 replied:
You can't use any CMake variables in conf.py.  It has to be a conf.py.in (which don't work with RTD).

agcapps replied:
thanks.

kennyweiss commented on src/conf.py:
```diff
@@ -29,10 +32,27 @@
+    'sphinx.ext.todo',
+    'sphinx.ext.coverage',
+    'sphinx.ext.mathjax',
+    'breathe'
```
Is there any reason to keep the dependency on `breathe` (which pull doxygen into sphinx)?  
If not, now might be a good time to remove it.

white238 replied:
5 seconds of googling showed there is a way to add breathe to RTD but I vote to remove it until we decide its worth doing.

rhornung67 replied:
I don' think we need it. Using breathe to pull doxygen into sphinx results in source code docs that are less useful; i.e., not as easily navigable, etc.

agcapps replied:
I'll take out loading `breathe` and all lines having to do with it.

kennyweiss commented on src/conf.py:
```diff
@@ -285,3 +311,11 @@
+
+breathe_projects = { "axomdoc": "@CMAKE_CURRENT_BINARY_DIR@/../doxygen/xml/" }
+# slam has the variant:
+#breathe_projects = { "slam": "../doxygen/xml/" }
```
It should be safe to remove this.

kennyweiss commented on src/docs/index.rst:
```diff
@@ -55,13 +58,16 @@ for Axom software components:
+   Lumberjack (Scalable parallel message logging and filtering) <../axom/lumberjack/docs/sphinx/index>
+   Sidre (Simulation data repository) <../axom/sidre/docs/sphinx/index>
+   Slam (Set-theoretic lightweight API for meshes) <../axom/slam/docs/sphinx/index>
+   Slam (Set-theoretic lightweight API for meshes) <../axom/quest/docs/sphinx/index>
```
Typo: the second `Slam` should be `Quest`

agcapps replied:
Thanks!

kennyweiss commented on src/docs/index.rst:
```diff
@@ -55,13 +58,16 @@ for Axom software components:
+.. toctree::
+   :maxdepth: 1
+
+   Slic (Simple Logging Interface Code) <../axom/slic/docs/sphinx/index>
```
Should these be alphabetized?

gzagaris commented on src/docs/index.rst:
```diff
@@ -55,13 +58,16 @@ for Axom software components:
+   Lumberjack (Scalable parallel message logging and filtering) <../axom/lumberjack/docs/sphinx/index>
+   Sidre (Simulation data repository) <../axom/sidre/docs/sphinx/index>
+   Slam (Set-theoretic lightweight API for meshes) <../axom/slam/docs/sphinx/index>
+   Slam (Set-theoretic lightweight API for meshes) <../axom/quest/docs/sphinx/index>
```
I think you meant quest here

agcapps replied:
I did.  Thanks for catching this.

gzagaris commented on src/docs/index.rst:
```diff
@@ -55,13 +58,16 @@ for Axom software components:
-  *  `Quest (Querying on surface tool) <../../quest_docs/html/index.html>`_
-  *  `Mint (Mesh data model) <../../mint_docs/html/index.html>`_
-  *  `Primal (Computational geometry primitives) <../../primal_docs/html/index.html>`_
+.. toctree::
```
I think it would look nicer if we use captions for the different `toctrees`

agcapps replied:
Perfect!  Thank you @gzagaris .

agcapps commented:
Apologies everyone: I meant to title this PR "Unified Sphinx into one."  We already had a unified doxygen.

white238 commented on src/docs/index.rst:
```diff
@@ -39,10 +39,13 @@ The main drivers of the Axom project are to:
+The Axom Quickstart Guide contains information about accessing the code,
+configuring and building, linking with an application, etc.
 
+.. toctree::
```
Did the link not work? This looked nicer the other way imo.

agcapps replied:
The link did work.  But I wanted the toctree caption, mainly because of the way it shows up in the side bar, and I did not want two links to the quickstart very close to each other.

white238 commented on src/docs/index.rst:
```diff
@@ -55,13 +58,16 @@ for Axom software components:
+   Primal (Computational geometry primitives) <../axom/primal/docs/sphinx/index>
 
 --------------------------
 Source Code Documentation
```
None of these links work.

agcapps replied:
fixed

agcapps commented:
> @agcapps it looks like the links in the "Further documentation" section are broken. Also, please change the title on the main page from:
> 
> Axom: A set of computer science tools/components that will be the basis of future next generation and current ASC simulation codes at LLNL.
> 
> to something like:
> 
> Axom provides a set of computer science tools/components that can be used as the foundation of a HPC simulation code.

@rhornung67 , @white238 , I fixed the links.  Rich, I updated the title line.

rhornung67 commented:
👍

agcapps commented:
I think I got everyone's recommendations.  Now I have to figure out how to make the RTD theme work.

agcapps commented:
OK, this version of the Sphinx docs are visible at https://axom.readthedocs.io/en/feature-capps2-single-doxygen/.  I didn't realize that readthedocs does not build the Doxygen as well, so all those links will be dead, but the other stuff should be working.

Thanks for your comments.

gzagaris commented:
Look really nice! 

One minor improvement would be to shorten the caption for the last `toctree` so that it fits within the sidebar. Perhaps, just "Developer Resources"
